### PR TITLE
Fixes #747 - Shorten method names in AccumuloClient builder

### LINKED ISSUE #747: Shorten method names in AccumuloClient builder
[AccumuloClient](https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java) has long method names for its builder that could be shortened before 2.0 is released.

Below are examples of this proposed shortening:

```java
// current
Accumulo.newClient().forInstance("myinstance", "zookeeper1,zookeeper2")
      .usingPassword("myuser", "mypassword").build();
// proposed
Accumulo.newClient().instance("myinstance", "zookeeper1,zookeeper2")
      .user("myuser", "mypassword").build();



// current
Accumulo.newClient().usingClientInfo(clientInfo)
     .usingToken("user, new PasswordToken("pass")).build();
// proposed
Accumulo.newClient().from(clientInfo).user("user", new PasswordToken("pass")).build();
```

Below is a chart of the proposed method names for the [AccumuloClient](https://github.com/apache/accumulo/blob/master/core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java) builder. Naming is subjective so I am open to no change or different names. Feel free to comment on this issue:

| Current Method Name | Proposed Method Name |
|--------------------------|-------------------------------|
|  forInstance(String name, String zk) | instance(String name, String zk) |
| usingProperties(String propsFilePath) | from(String propsFilePath) |
| usingProperties(Properties properties) | from(Properties props) |
| usingClientInfo(ClientInfo info) | from(ClientInfo info) |
| usingPassword(String username, CharSequence password) | user(String username, CharSequence password) |
| usingKerberos(String principal, String keytabFile) | kerberosUser(String principal, String keytabPath) |
| usingToken(String principal, AuthenticationToken token) | user(String principal, AuthenticationToken token) |
| withTruststore | truststore |
| withKeystore | keystore |
| withJsse | jsse |
| withPrimary | primary |
| withQop | qop |
| withSsl | ssl |
| withSasl | sasl |
| withBatchWriterConfig | batchWriterConfig |
| withBatchScannerQueryThreads | batchScannerQueryThreads |
| withScannerBatchSize | scannerBatchSize |

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 1]
```diff
@@ -377,7 +378,7 @@ AccumuloClient changeUser(String principal, AuthenticationToken token)
      *          Path to properties file
      * @return this builder
      */
-    AccumuloClientFactory usingProperties(String propertiesFile);
+    AccumuloClientFactory from(CharSequence propertiesFile);
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: I think this should be Path or File (probably Path).

File: test/src/main/java/org/apache/accumulo/test/ShellServerIT.java [hunk 2]
```diff
@@ -176,7 +175,7 @@ public String getErrorMessage() {
 
     TestShell(String user, String rootPass, String instanceName, String zookeepers, File configFile)
         throws IOException {
-      ClientInfo info = Accumulo.newClient().from(configFile.getAbsolutePath()).info();
+      ClientInfo info = ClientInfo.from(configFile.toPath());
```
Comment (ctubbsii) [inline test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: Are these equivalent? I'm not 100% sure the relationship between File and Path, and when to use one or the other, or how absolute/relative paths get resolved in either.

Comment (mikewalch) [reply test/src/main/java/org/apache/accumulo/test/ShellServerIT.java]: They should be. I ran this IT and it passes.

File: core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java [hunk 3]
```diff
@@ -363,14 +364,24 @@ AccumuloClient changeUser(String principal, AuthenticationToken token)
    * Builder method for setting Accumulo instance and zookeepers
    */
   interface InstanceArgs {
-    AuthenticationArgs forInstance(String instanceName, String zookeepers);
+    AuthenticationArgs to(CharSequence instanceName, CharSequence zookeepers);
   }
 
   /**
    * Builder methods for creating AccumuloClient using properties
    */
```
Comment (keith-turner) [inline core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: I think all of these inner interfaces need a since tag

Comment (mikewalch) [reply core/src/main/java/org/apache/accumulo/core/client/AccumuloClient.java]: Fixed in 23da39e93c3209

**Pull request merged**